### PR TITLE
Move observability, add test helper for observability and record metrics in loggingexporter.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Comment by sjkaris** [comment]
Thanks!

File: exporter/exportertest/exportertest.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// limitations under the License.
+
+package exportertest
+
+import (
+	"encoding/json"
+)
+
+// ToJSON marshals a generic interface to JSON to enable easy comparisons.
+func ToJSON(v interface{}) []byte {
```
Comment (pjanotti) [inline exporter/exportertest/exportertest.go]: nit: I am not sure if it is worth to keep this file and function

File: observability/observabilitytest/observabilitytest.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+type nopMetricsExporter int
+
+var _ view.Exporter = (*nopMetricsExporter)(nil)
+
+func (cme *nopMetricsExporter) ExportView(vd *view.Data) {}
+
+// CheckRecordedMetricsForTraceExporter checks that the given TraceExporter records the correct set of metrics with the correct
+// set of tags by sending some TraceData to the exporter. The exporter should be able to handle the requests.
+func CheckRecordedMetricsForTraceExporter(t *testing.T, te exporter.TraceExporter, exporterTagName string) {
```
Comment (pjanotti) [inline observability/observabilitytest/observabilitytest.go]: Nice!

File: exporter/exportertest/exportertest.go (hunk 1 above)
Comment (bogdandrutu) [reply exporter/exportertest/exportertest.go]: Will remove in a separate PR.

File: observability/observabilitytest/observabilitytest.go (hunk 2 above)
Comment (bogdandrutu) [reply observability/observabilitytest/observabilitytest.go]: 👍

**Pull request merged**